### DISCUSSION TITLE
FormRequest::validated() does not work with an array of objects

### DISCUSSION THREAD

**Discussion by develgold** [laravel/framework#40884] [Q&A] [answered]
- Laravel Version: 9.0
- PHP Version: 8.1.0

### Description:
FormRequest::validated() does not work with an array of objects

### Steps To Reproduce:
Request class:
`
class UserRequest extends FormRequest {
    public function rules()
    {
        return [
            'name' => 'string',
            'items.*.id' => 'integer'
        ];
    }
}
`
Input:
`{"name": "Bob", "items": [{"id": 1}]}`

$request->validated() returns only name

**Comment by hmennen90** [comment, marked as answer]
@Develgold Just tested your example

```
public function testValidatedMethodShowsAllValidatedFields() {
        $data = [
            'name' => 'string',
            'items' => [
                [
                    'id' => 1
                ]
            ],
            'not_shown_key' => 'myString'
        ];

        $rules = [
            'name' => 'string',
            'items.*.id' => 'integer'
        ];

        $trans = $this->getIlluminateArrayTranslator();

        $validator = new Validator($trans, $data, $rules);

        $result = $validator->validated();

        $this->assertArrayHasKey('name', $result);

        $this->assertArrayHasKey('items', $result);

        $this->assertArrayHasKey('id', $result['items'][0]);
    }
```


This test does not Fail.

What did you get, if you call $request->all() ? Maybe it's a typo

**Reply by develgold** [reply]
It's a typo, thanks

**Comment by hmennen90** [comment]
Ok - got it:

Your JSON casted to an Array results only the first level in array => name and items - but every element in items remains as Object:

```
[
  "name" => "Bob"
  "items" => array:1 [
    0 => {#1844
      +"id": 1
    }
  ]
]
```
